controllers/argocd/configmap.go

```diff
@@ -680,15 +680,17 @@ func (r *ReconcileArgoCD) reconcileSSHKnownHosts(cr *argoprojv1a1.ArgoCD) error
 // reconcileTLSCerts will ensure that the ArgoCD TLS Certs ConfigMap is present.
 func (r *ReconcileArgoCD) reconcileTLSCerts(cr *argoprojv1a1.ArgoCD) error {
 	cm := newConfigMapWithName(common.ArgoCDTLSCertsConfigMapName, cr)
-	cm.Data = getInitialTLSCerts(cr)
```

Comment:
where are the changes in configmap.go coming from?

Reply:
There are changes to reconciler logic.  What amount of testing does it come with?

Reply:
I am equally surprised. I dont know why this unit test was failing. 

The issue is coming from isObjectFound method but I dont see any changes made. 
https://github.com/argoproj-labs/argocd-operator/blob/2da8dac3bba7f0fbe52cd390dd10120d5545da34/controllers/argoutil/resource.go#L86

Reply:
To be honest there is no change in the logic. It's mostly refactoring. So according to me no additional testing is needed.

Reply:
Which unit test was failing?
The change of logic/behavior is that now it only sets controller owner reference when object does not exist (or when a new cm is created).

Reply:
This is the failing unit test
`TestReconcileArgoCD_reconcileTLSCerts_withUpdate`

However, I figured out the root cause of the issue. 
`The latest version of controller-runtime overrides the object instead of updating`.

Please refer to the below PR.
https://github.com/kubernetes-sigs/controller-runtime/pull/1651

Reply:
> The change of logic/behavior is that now it only sets controller owner reference when object does not exist (or when a new cm is created).

That makes sense as existing objects(configmaps) would already have the owner reference set.

Reply:
https://github.com/kubernetes-sigs/controller-runtime/pull/1651 is a fix to the fake client.  The question is that is the current reconciler not doing the right thing and needs to be changed? If the fix to the fake client only affect testing, the failing test should be fixed.